### PR TITLE
Reimplements the enable/disable functionality in pure js

### PR DESCRIPTION
**Pull request by mko-sci**
The reimplementation was necessary due to the filter options which were also added to the url, based on whatever option was chosen within the process view on the left side panel.

Fixes: [SIRI-608](https://scireum.myjetbrains.com/youtrack/issue/SIRI-608/Process-Refresh-gn%C3%A4diger-machen)

### CODE REVIEW

File: src/main/resources/default/templates/biz/process/process.html.pasta [hunk 1]
```diff
@@ -145,11 +155,41 @@
         <i:render name="body"/>
     </t:sidebar>
 
-    <i:if test="process.shouldAutorefresh() && autoRefresh">
+    <i:if test="process.shouldAutorefresh()">
         <script type="text/javascript">
             sirius.ready(function () {
-                setTimeout(function () {
-                    window.location.reload();
+                let autoRefreshStorageVariableName = "ps.isAutoRefreshEnabled";
```
Comment (sabieber) [inline src/main/resources/default/templates/biz/process/process.html.pasta:161]:
```suggestion
                const autoRefreshStorageVariableName = "sirius.ps.isAutoRefreshEnabled";
```

File: src/main/resources/default/templates/biz/process/process.html.pasta [hunk 2]
```diff
@@ -145,11 +155,41 @@
...
+                let isAutoRefreshEnabled = sirius.readLocalStore(autoRefreshStorageVariableName) === "true" ? true : false;
+
+                let _enableAutoRefresh = document.getElementById("enableAutoRefresh");
+                let _disableAutoRefresh = document.getElementById("disableAutoRefresh");
```
Comment (sabieber) [inline src/main/resources/default/templates/biz/process/process.html.pasta:165]:
```suggestion
                const _enableAutoRefresh = document.getElementById("enableAutoRefresh");
                const _disableAutoRefresh = document.getElementById("disableAutoRefresh");
```

**Review by sabieber — CHANGES_REQUESTED** [review]

**Review by sabieber — APPROVED** [review]

File: src/main/resources/default/templates/biz/process/process.html.pasta [hunk 3]
```diff
@@ -145,11 +155,41 @@
...
+                const _enableAutoRefresh = document.getElementById("enableAutoRefresh");
+                const _disableAutoRefresh = document.getElementById("disableAutoRefresh");
+
+                if (isAutoRefreshEnabled === true) {
+                    _disableAutoRefresh.classList.remove("d-none");
+                } else {
+                    _enableAutoRefresh.classList.remove("d-none");
+                }
+
+                _enableAutoRefresh.onclick = function (event) {
```
Comment (andyHa) [inline src/main/resources/default/templates/biz/process/process.html.pasta:173]:
```suggestion
                function enableAutoRefresh() {
```

File: src/main/resources/default/templates/biz/process/process.html.pasta [hunk 4]
```diff
@@ -145,11 +155,41 @@
...
+                    event.preventDefault();
+                    isAutoRefreshEnabled = true;
+                    sirius.writeLocalStore(autoRefreshStorageVariableName, true);
+                    _enableAutoRefresh.classList.add("d-none");
+                    _disableAutoRefresh.classList.remove("d-none");
+                };
+
+                _disableAutoRefresh.onclick = function (event) {
```
Comment (andyHa) [inline src/main/resources/default/templates/biz/process/process.html.pasta:181]: s.a.

File: src/main/resources/default/templates/biz/process/process.html.pasta [hunk 5]
```diff
@@ -23,6 +22,17 @@
                 </span>
             </i:block>
             <i:block name="actions">
+                <i:if test="process.shouldAutorefresh()">
+                    <a id="disableAutoRefresh"
+                       class="btn btn-link d-none"
+                       href="#">
```
Comment (andyHa) [inline src/main/resources/default/templates/biz/process/process.html.pasta:28]:
```suggestion
                       href="javascript:enableAutoRefresh()">
```

Comment (andyHa) [reply src/main/resources/default/templates/biz/process/process.html.pasta:28]: less magic...more obvious (or via click="...")

**Review by andyHa — CHANGES_REQUESTED** [review]

File: src/main/resources/default/templates/biz/process/process.html.pasta (hunk 5 above)
Comment (qw3ry) [reply src/main/resources/default/templates/biz/process/process.html.pasta:28]: Disadvantage: You have to put `enableAutoRefresh()` in the global js context

**Comment by mko-sci** [comment]
Nach persönlicher Absprache mit AHA: Kann so bleiben. Wird jetzt also gemerged.

**Pull request merged**